tests/unit/test_fields.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+from unittest.mock import MagicMock, call
+
+from pytest_splunk_addon.standard_lib.addon_parser.fields import (
+    Field,
+    convert_to_fields,
+)
+
+NAME = "name"
+NAME1 = "Name1"
+TYPE = "type"
+TYPE1 = "Type1"
+MULTI_VALUE = "multi_value"
+MULTI_VALUE1 = "Multi_value1"
+EXPECTED_VALUES = "expected_values"
+EXPECTED_VALUES1 = "Expected_values1"
+NEGATIVE_VALUES = "negative_values"
+NEGATIVE_VALUES1 = "Negative_values1"
+CONDITION = "condition"
+CONDITION1 = "Condition1"
+VALIDITY = "validity"
+VALIDITY1 = "Validity1"
+
+PROPERTIES = f"""{NAME1}
+{TYPE}={TYPE1}
+{MULTI_VALUE}={MULTI_VALUE1}
+{CONDITION}={CONDITION1}
+{VALIDITY}={VALIDITY1}
+{EXPECTED_VALUES}={EXPECTED_VALUES1}
+{NEGATIVE_VALUES}={NEGATIVE_VALUES1}"""
+
+
+@pytest.fixture()
+def field_json():
+    return {
+        NAME: NAME1,
+        TYPE: TYPE1,
+        MULTI_VALUE: MULTI_VALUE1,
+        EXPECTED_VALUES: EXPECTED_VALUES1,
+        NEGATIVE_VALUES: NEGATIVE_VALUES1,
+        CONDITION: CONDITION1,
+        VALIDITY: VALIDITY1,
+    }
+
+
+@pytest.fixture
+def default_field(field_json):
+    return Field(field_json=field_json)
+
+
+@pytest.fixture
+def field_mock(monkeypatch):
+    field_mock = MagicMock()
+    monkeypatch.setattr(
+        "pytest_splunk_addon.standard_lib.addon_parser.fields.Field", field_mock
+    )
+    return field_mock
+
+
+def test_field_constructor(field_json):
+    field = Field(field_json=field_json)
+    assert field.name == NAME1
+    assert field.type == TYPE1
+    assert field.multi_value == MULTI_VALUE1
+    assert field.expected_values == EXPECTED_VALUES1
+    assert field.negative_values == NEGATIVE_VALUES1
+    assert field.condition == CONDITION1
+    assert field.validity == VALIDITY1
+
+    for key in [
+        TYPE,
+        MULTI_VALUE,
+        EXPECTED_VALUES,
+        NEGATIVE_VALUES,
+        CONDITION,
+        VALIDITY,
+    ]:
+        field_json.pop(key)
+
+    field = Field(field_json=field_json)
+    assert field.name == NAME1
+    assert field.type == "required"
+    assert field.multi_value is False
+    assert field.expected_values == ["*"]
+    assert field.negative_values == ["-", ""]
+    assert field.condition == ""
+    assert field.validity == NAME1
+
+
+def test_str(default_field):
+    assert str(default_field) == NAME1
+
+
+def test_get_type(default_field):
+    assert default_field.get_type() == TYPE1
+
+
+def test_parse_fields(default_field, field_mock):
+    list(default_field.parse_fields([{"one": 1}, {"two": 2}], three=3, four=4))
```

Comment:
I'm just wondering shall we check also what is returned?
```suggestion
    result = list(default_field.parse_fields([{"one": 1}, {"two": 2}], three=3, four=4))
    assert result == [field_mock.return_value] * 2
```

Reply:
Done, it is better to do it I think. Initially I didn't do it because it returns mock called result